optuna/study/study.py

```diff
@@ -536,7 +534,7 @@ def ask(
         }
 
         # Sync storage once every trial.
-        self._thread_local.cached_all_trials = None
+        self._cached_all_trials.set(None)
```

Comment:
Could you please create a new context for each `ask()` method call instead of clearing the cache?

Reply:
Hey @c-bata,

It won't work. Object `Study` is shared between threads. Once we create a new `_cached_all_trials` in one thread, all threads lose data in context.
Test `test_cache_is_invalidated` works fine for n_jobs=1 and fails for n_jobs>1 at `assert study._cached_all_trials.get() is not None`

Another possible solution - clone `Study` for each thread